api/src/neo4j/queries/gene.js

```diff
@@ -46,5 +47,94 @@ RETURN apoc.map.mergeList(COLLECT(value.data)) as gene
   return { ...gene, externalDbs: reformatExternalDbs(gene.externalDbs) };
 };
 
+const getGenesForHPA = async () => {
+  const statement = `
+MATCH (g:Gene:HumanGem)-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(s:Subsystem)-[:V1_3_0]-(ss:SubsystemState)
+USING JOIN ON r
+RETURN DISTINCT [g.id, ss.name, s.id]
+`;
+
+  return queryListResult(statement);
+};
+
+const BASE_URL = 'https://metabolicatlas.org';
+
+const getGeneDetailsForHPA = async ({ id }) => {
+  const statement = `
+MATCH (:Gene:HumanGem {id:'${id}'})-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(s:Subsystem)-[:V1_3_0]-(ss:SubsystemState)
+WITH DISTINCT s.id as sids, COLLECT(r.id) as rids, ss
+UNWIND sids as sid
+CALL apoc.cypher.run("
+  MATCH (:Subsystem {id: $sid})-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(cm:CompartmentalizedMetabolite)
+  USING JOIN ON r
+  WITH DISTINCT cm
+  MATCH (m:Metabolite)-[:V1_3_0]-(cm)-[:V1_3_0]-(c:Compartment)-[:V1_3_0]-(cs:CompartmentState)
+  USING JOIN ON c
+  RETURN DISTINCT { id: $sid, compartments: COLLECT(DISTINCT(cs.name)), model_metabolite_count: COUNT(DISTINCT(m.id)), compartment_metabolite_count: COUNT(DISTINCT(cm.id)) } as data
+
+  UNION
+
+  MATCH (:Subsystem {id: $sid})-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(g:Gene)
+  USING JOIN ON r
+  RETURN DISTINCT { id: $sid, genes: COLLECT(DISTINCT(g.id)) } as data
+
+  UNION
+
+  MATCH (:Subsystem {id: $sid})-[:V1_3_0]-(r:Reaction)
+  RETURN DISTINCT { id: $sid, reaction_count: COUNT(DISTINCT(r.id)) } as data
+
+  UNION
+
+  MATCH (:Subsystem {id: $sid})-[:V1_3_0]-(r:Reaction)
+  WHERE r.id IN $rids
+  RETURN DISTINCT { id: $sid, reactions_catalysed: COUNT(DISTINCT(r)) } as data
+
+  UNION
+
+  MATCH (:Subsystem {id: $sid})-[:V1_3_0]-(ssvg:SvgMap)
+  RETURN { id: $sid,  svgs: COLLECT(DISTINCT(ssvg.filename)) } as data
```

Comment:
There is, and there will always be, a single SVG map per subsystem. Is it really needed to have `DISTINCT` here?

Reply:
Thanks! Addressed in: https://github.com/MetabolicAtlas/MetabolicAtlas/pull/546/commits/e3f6ac9e8e5d84decd379fdd3fbe4829cbe1b61a